pkgs/top-level/python-packages.nix

```diff
@@ -4657,6 +4657,10 @@ in {
 
   seqdiag = callPackage ../development/python-modules/seqdiag { };
 
+  sequoia = toPythonModule (pkgs.sequoia.override {
+    pythonSupport = true;
```

Comment:
`python3Packages = python.pkgs;`

How about Python 2 or PyPy? Is it supported? if not, check the `disabledIf` in this file.

Reply:
sequoia is not available in PyPi and is not supposed to be supported by Python 2. I've encountered that in the `opencv` definitions. There, the variable passed to the derivation is named `pythonPackages`, not `python3Packages`. Should I switch to that as well? Another question: should I disable `isPyPy` **and** `isPy27` with `disabledIf`? Or would it be better to not pass this variable but only pass and only use `python3Packages = python.pkgs;` in `sequoia`'s `default.nix`. There seem to be so many options..

Reply:
Yes, `disabledIf (isPyPy || !isPy3k) toPythonModule ...` should be sufficient.

There are indeed a lot of options. Packaging is difficult, so we try to simplify it with builders for specific ecosystems. Not everything fits in, e.g. polyglot packages like these that provide bindings. As we gain more experience with packaging such packages we can start documenting best practices regarding those as well and try to converge the existing expressions to conform to those best practices.

Last week I looked at how the package set changed in the last 5 or so years. Back then, many Python packages were defined in `all-packages.nix`. Sometimes they were Python 2, and sometimes a Python 3 version. Bindings were generated sometimes, and if they were, you had to figure out yourself for which Python version they were built and hope they were actually compatible. We've come a long way since but there's definitely room for improvement!

Reply:
Cool. I'll use `disabledIf (isPyPy || !isPy3k)`.

What about the arguments to override? Should I pass `python3Packages = self`? For comparison, in `opencv`s' `toPythonModule`, an argument named `pythonPackages = self` is always passed. AFAIK, only old versions of opencv support _only_ Python2 so does it mean that in NixOS, _all_ compiled versions of opencv use Python2 by default?.

I tried renaming `sequoia`'s `python3Packages` to `pythonPackages` and pass `pythonPackages = self`, just like in `opencv`'s. But then, `nix-build -A sequoia` started using Python2's packages.. I could override that in `all-packages.nix` as well, just like [`afew`](https://github.com/NixOS/nixpkgs/blob/e7e24af6eeaaa3361eadc91d8e07b4812f346cd2/pkgs/top-level/all-packages.nix#L495) does, but It seems kind of weird..

Reply:
@FRidh I've made a decision and pushed it. Please tell me your opinion and so we could continue on merging this.

Reply:
Looks good.